com.unity.cinemachine/Runtime/Behaviours/CinemachineFreeLook.cs

```diff
@@ -439,40 +439,41 @@ float GetYAxisClosestValue(Vector3 cameraPos, Vector3 up)
                 }
                 dir.x = 0;
 
-                // Sample the spline in a few places, find the 2 closest, and lerp
-                int i0 = 0, i1 = 0;
-                float a0 = 0, a1 = 0;
-                const int NumSamples = 13;
-                float step = 1f / (NumSamples-1);
-                for (int i = 0; i < NumSamples; ++i)
-                {
-                    float a = Vector3.SignedAngle(
-                        dir, GetLocalPositionForCameraFromInput(i * step), Vector3.right);
-                    if (i == 0)
-                        a0 = a1 = a;
-                    else
-                    {
-                        if (Mathf.Abs(a) < Mathf.Abs(a0))
-                        {
-                            a1 = a0;
-                            i1 = i0;
-                            a0 = a;
-                            i0 = i;
-                        }
-                        else if (Mathf.Abs(a) < Mathf.Abs(a1))
-                        {
-                            a1 = a;
-                            i1 = i;
-                        }
-                    }
-                }
-                if (Mathf.Sign(a0) == Mathf.Sign(a1))
-                    return i0 * step;
-                float t = Mathf.Abs(a0) / (Mathf.Abs(a0) + Mathf.Abs(a1));
-                return Mathf.Lerp(i0 * step, i1 * step, t);
+                // We need to find the minimum of the angle of function using steepest descent
+                return SteepestDescent(0f, 1f, dir);
```

Comment:
Very nice algorithm.  However, it fails for this type of orbit configuration:

![image](https://user-images.githubusercontent.com/6424658/191979761-63a48605-d14e-4e7b-9bd3-6ff0a9d94e06.png)

Reply:
It's possible that there's no simple solution that solves all cases.  The old algo also failed in this case, so your fix is an improvement.

Reply:
Maybe an easy fix is to start with a better first guess, perhaps based on the spline waypoint that's closest to the camera